index.d.ts

```diff
@@ -2910,23 +2910,23 @@ declare namespace Eris {
         members?: Collection<Omit<Member, "user" | "deaf" | "mute">>;
         roles?: Collection<Role>;
         channels?: Collection<PartialChannel>;
-        messages?: Collection<Message>;
+        messages?: Collection<Message<T>>;
       };
       options?: InteractionDataOptions[];
     };
-    guildID?: string;
-    member?: Member;
+    guildID: T extends GuildTextableWithThread ? string : never;
+    member: T extends GuildTextableWithThread ? Member : never;
```

Comment:
How would this be displayed if TS can't figure out if it's a dm channel or guild channel

Reply:
it doesn't display at all due to the never
there really isn't a better solution besides not having a generic because an explicit undefined is wrong

Reply:
on further inspection, this does not work as expected in any scenario

GuildTextable extends Textable so any provided value will show the guildID & member as being present

Reply:
Will try to get some more feedback on this